src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java

```diff
@@ -390,6 +430,12 @@ public ElasticsearchSinkConnectorConfig(Map<String, String> props) {
     super(CONFIG, props);
   }
 
+  public Map<String, Object> sslConfigs() {
+    ConfigDef sslConfigDef = new ConfigDef();
+    addClientSslSupport(sslConfigDef);
+    return sslConfigDef.parse(originalsWithPrefix(CONNECTION_SSL_CONFIG + "."));
```

Comment:
It's not clear to me why this approach is used, rather than something like:
```
    return originalsWithPrefix(CONNECTION_SSL_CONFIG + ".");
```
The ConfigDef for the class already contains these, so why do we need to create a new ConfigDef to parses the originals?

Reply:
Using just `originalsWithPrevis(...)`,  not parse the values into the correct type or use the default value (it pulls "originals" but we want "values"). There are two functions to pull values-with-prefix in AK, but neither is really suited for this case because they both assume that the sub-field is also defined on the main ConfigDef, which we don't want to do in this connector since it will collide with names for connecting to the rest of CP. 

I wasn't thrilled about this strategy, and I think the cleanest way to handle without adding another `withPrefix`-parsing function to AK, which we can't rely on.

Reply:
Ok, that's fair. Another approach would be to find the configs with the prefix, get the value for each of those configs, and add to a new map without the prefixed name. Is it really simpler, clearer and better to reparse and revalidating the original strings?

Reply:
I think this line is very readable: `return sslConfigDef.parse(originalsWithPrefix(CONNECTION_SSL_CONFIG_PREFIX));`. It is perhaps redundant to parse it twice, but it should be a fast operation that only happens when connector task starts. To make this clearer, I would rather make a static property of `sslConfigDef` so it's not goofily created multiple times. Here we're relying on AK abstractions and intentions instead of prescriptively repeating work. WDYT?